core/src/robot/AbstractTransmission.cc

```diff
@@ -0,0 +1,417 @@
+        hresult_t returnCode = hresult_t::SUCCESS;
+
+        if (!isAttached_)
+        {
+            PRINT_ERROR("Transmission not attached to any robot. Impossible to refresh proxies.");
+            returnCode = hresult_t::ERROR_INIT_FAILED;
+        }
+
+        auto robot = robot_.lock();
+        if (returnCode == hresult_t::SUCCESS)
+        {
+            if (!robot)
+            {
+                PRINT_ERROR("Robot has been deleted. Impossible to refresh proxies.");
+                returnCode = hresult_t::ERROR_GENERIC;
+            }
+        }
+
+        if (returnCode == hresult_t::SUCCESS)
+        {
+            if (!isInitialized_)
+            {
+                PRINT_ERROR("Transmission not initialized. Impossible to refresh proxies.");
+                returnCode = hresult_t::ERROR_INIT_FAILED;
+            }
+        }
+
+        if (returnCode == hresult_t::SUCCESS)
+        {
+            if (!robot->getIsInitialized())
+            {
+                PRINT_ERROR("Robot not initialized. Impossible to refresh proxies.");
+                returnCode = hresult_t::ERROR_INIT_FAILED;
+            }
+        }
+
+        for (unsigned int i = 0; i < jointNames_.size(); i++)
+        {
+            if (returnCode == hresult_t::SUCCESS)
+            {
+                // jointModelIndices_ is empty at this point, therefore inject a 0 and later get the right index
+                jointModelIndices_.push_back(0);
+                returnCode = ::jiminy::getJointModelIdx(robot->pncModel_, jointNames_[i], jointModelIndices_[i]);
+            }
+        }
+
+
+        for (unsigned int i = 0; i < jointNames_.size(); i++)
+        {
+            joint_t tmp;
+            if (returnCode == hresult_t::SUCCESS)
+            {
+                // jointTypes_ is empty at this point, therefore inject a 0 and later get the right type
+                jointTypes_.push_back(tmp);
+                returnCode = getJointTypeFromIdx(robot->pncModel_, jointModelIndices_[i], jointTypes_[i]);
+            }
+        }
+
+        for (unsigned int i = 0; i < jointNames_.size(); i++)
+        {
+            if (returnCode == hresult_t::SUCCESS)
+            {
+                // Transmissions are only supported for linear and rotary joints
+                if (jointTypes_[i] != joint_t::LINEAR && jointTypes_[i] != joint_t::ROTARY && jointTypes_[i] != joint_t::ROTARY_UNBOUNDED)
+                {
+                    PRINT_ERROR("A transmission can only be associated with a 1-dof linear or rotary joint.");
+                    returnCode = hresult_t::ERROR_BAD_INPUT;
+                }
+            }
+        }
+
+        // Populate motorIndices_
+        std::weak_ptr<AbstractMotorBase const> motor;
+        for (std::string const & motorName : getMotorNames())
+        {
+            if (returnCode == hresult_t::SUCCESS)
+            {
+                returnCode = robot->getMotor(motorName, motor);
+                auto motorTemp = motor.lock();
```

Comment:
for you `motorTmp` is better ? or you just do not like tmp at all ? the problem is that i do not want to call it `motor_` (this is something else) and also i cannot call it `motor` (what I usually do for temporary variables) bc this is already used here

Reply:
Tmp or similar must be avoided because it does not bring any information about the intend or what it is. Here it is not easy. Eventually `motorLocked`, `motorPtr`, or even `motorWeakPtr`. it is not perfect but carry some information even though limited.

Reply:
Ok I see the point